README.md

```diff
@@ -229,6 +245,36 @@ Stats for both
 ./metagraph stats -a annotation.column.annodbg graph.dbg
 ```
 
+## Developer Notes
+
+### Makefile
+
+The `Makefile` in the top level source directory can be used to build and test `metagraph` more conveniently. The following
+arguments are supported:
+* `env`: environment in which to compile/run, if empty on the host, if `docker` in a docker container
+* `alphabet`: compiling metagraph for a certain alphabet (e.g. `DNA` or `Protein`)
+* `additional_cmake_args`: additional arguments to pass to cmake.
+
+Examples:
+
+```
+# compiles metagraph in a docker container for the `Protein` alphabet
+make build-metagraph env=docker alphabet=Protein
+```
+
+### Developing with Docker Images
+
+To have a portable and consistent developer environment, docker containers can be used.
+
+The Dockerfile in the source root consists of the following 3 stages:
+1. `metagraph_dev_env`: contains all dependencies to build metagraph. Can also be used during development by mounting the code base and build dir on the host (this is done in `make build-metagraph env=docker`)
+2. `metagraph_bin`: based on `docker_dev_env` but contains the `metagraph` binary. It is more of an intermediary image and not really used per se.
```

Comment:
Does anyone really need to know about these two, if it's just a detail (like, to break things into smaller components / make the build faster)?

Reply:
It's useful to know what `metagraph_dev_env` means as it is used in the `Makefile`. I'll see if I can move it to a different location though.

Reply:
I mean, we have a bunch of intermediate targets in cmake config as well, but they are mostly auxiliary and there is no need to describe them in the readme.
Those who need them will almost surely go through the config file and find them anyway.
And those who don't need these might just get confused by the variaty...

Reply:
Maybe we can just put these descriptions as comments in the Makefile. What do you think?